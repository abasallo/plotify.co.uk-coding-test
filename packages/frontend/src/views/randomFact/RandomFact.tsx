import { useEffect, useState } from 'react'

import { fetchRandomFact } from '../../services/randomFact'

import { withMainContainer } from '../../templates/withMainContainer'

const RandomFact = () => {
  const [randomFact, setRandomFact] = useState<string>('')

  useEffect(() => {
    const dataFetch = async () => setRandomFact(await fetchRandomFact())
    dataFetch()
  }, [])

  return <h1>{randomFact}</h1>
}

export default withMainContainer(RandomFact)
