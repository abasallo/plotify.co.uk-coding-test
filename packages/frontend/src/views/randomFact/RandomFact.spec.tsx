import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import RandomFact from './RandomFact'
import React from 'react'

describe('Random Facts View', () => {
  test('Renders properly', () => {
    expect(renderer.create(
      <BrowserRouter>
        <RandomFact />
      </BrowserRouter>
    ).toJSON()).toMatchSnapshot()
  })
})
