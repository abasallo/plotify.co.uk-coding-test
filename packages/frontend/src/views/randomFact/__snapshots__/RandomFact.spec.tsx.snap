// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Facts View Renders properly 1`] = `
<div
  className="MuiContainer-root sc-beqWaB dDnzPd MuiContainer-maxWidthLg"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-hLseeU csdnVv MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular sc-eDDNvR cPuiyT MuiToolbar-gutters"
    >
      <img
        className="sc-jTrPJq rmupg"
        onClick={[Function]}
        src="logo.png"
      />
    </div>
  </header>
  <div
    className="MuiContainer-root sc-gueYoa gcjcJu MuiContainer-maxWidthLg"
  >
    <h1 />
  </div>
  <div
    className="MuiBox-root MuiBox-root-1 sc-dmqHEX hWKGNj"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright Â© 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://www.linkedin.com/in/abasallo/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        alvaro@basallo.es
      </a>
       2023
    </p>
  </div>
</div>
`;
